### PR TITLE
update Menu and ItemsList to accept single child item

### PR DESCRIPTION
**Pull request by Tuxosaurus**
## Description

We're soon going to update the TopBar component and as I'm investigating what could be done I stumbled upon potential uses cases where we only have 1 entry in a menu, or at least that we render multiple items in a single component that **Menu** should accept.

```
<Menu>
    <ItemsRenderedThroughAnotherComponent />
</Menu>
```

## How it was tested

Locally on Firefox MacOS with canary release

### CODE REVIEW

File: src/itemsList/ItemsList.tsx [hunk 1]
```diff
@@ -38,6 +40,22 @@ export const ItemsList = (props: ItemsListProps) => {
   } = props
   const a11yAttrs = pickA11yProps<ItemsListProps>(props)
 
+  function isSingleChild(item: ChildrenType): item is ItemsListChild {
```
Comment (mcellier) [inline src/itemsList/ItemsList.tsx:43]: You might want to try something like:

```
const items = Array.isArray(children) ? children : [children]
const filteredChildren = children.filter(Boolean)
```

That way you won't have to keep the next 10 lines

File: src/itemsList/__snapshots__/ItemsList.unit.tsx.snap [hunk 2]
```diff
@@ -796,6 +796,326 @@ button:focus:not(.focus-visible).c1 {
 </ul>
 `;
 
+exports[`ItemsList Should render an unordered list with only one element 1`] = `
```
Comment (mcellier) [inline src/itemsList/__snapshots__/ItemsList.unit.tsx.snap:799]: nit: If it's not too long, you might want to migrate this to RTL.

**Review by mcellier — APPROVED** [review]

File: src/itemsList/ItemsList.tsx (hunk 1 above)
Comment (Tuxosaurus) [reply src/itemsList/ItemsList.tsx]: Done, I had try this but was misusing isArray. Works fine!

File: src/itemsList/__snapshots__/ItemsList.unit.tsx.snap (hunk 2 above)
Comment (Tuxosaurus) [reply src/itemsList/__snapshots__/ItemsList.unit.tsx.snap]: Done

**Review by gabrielagabriel — APPROVED** [review]
Great work.
Are you thinking about open a PR to another type of elements, example a list `ul` or a `div` instead?

**Comment by Tuxosaurus** [comment]
> Great work.
> Are you thinking about open a PR to another type of elements, example a list `ul` or a `div` instead?

For now I have no plan but I am thinking of looking at the usage of `nav` first.

**Pull request merged**